projects/plugins/crm/src/automation/workflow/class-workflow-repository.php

```diff
@@ -0,0 +1,244 @@
+	 * @return Automation_Workflow|false The workflow object or false if not found.
+	 */
+	public function find( int $id ) {
+		$row = $this->wpdb->get_row(
+			$this->wpdb->prepare( "SELECT * FROM {$this->table_name} WHERE id=%d", $id ), // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared, WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+			ARRAY_A
+		);
+
+		if ( ! is_array( $row ) ) {
+			return false;
+		}
+
+		return $this->map_row_to_workflow( $row );
+	}
+
+	/**
+	 * Get all workflows.
+	 *
+	 * @return Automation_Workflow[]
+	 */
+	public function find_all(): array {
+		$rows = $this->wpdb->get_results( "SELECT * FROM {$this->table_name}", ARRAY_A ); // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+
+		if ( empty( $rows ) ) {
+			return array();
+		}
+
+		foreach ( $rows as $index => $row ) {
+			$rows[ $index ] = $this->map_row_to_workflow( $row );
+		}
+
+		return $rows;
+	}
+
+	/**
+	 * Persist a workflow.
+	 *
+	 * This is used to both update and create a workflow.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param Automation_Workflow $workflow The workflow to persist.
+	 * @return void
+	 *
+	 * @throws Workflow_Exception Throw error if the workflow could not be persisted.
+	 */
+	public function persist( Automation_Workflow $workflow ): void {
+		if ( $workflow->get_id() && is_numeric( $workflow->get_id() ) ) {
+			$this->update( $workflow );
+		} else {
+			$this->insert( $workflow );
+		}
+	}
+
+	/**
+	 * Prepare the data to persist.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param Automation_Workflow $workflow The workflow to persist.
+	 * @return array The workflow raw data.
+	 */
+	protected function prepare_data_to_persist( Automation_Workflow $workflow ): array {
+		$data = $workflow->to_array();
+
+		$data['triggers']   = wp_json_encode( $data['triggers'] );
+		$data['steps']      = wp_json_encode( $data['steps'] );
+		$data['updated_at'] = gmdate( 'Y-m-d H:i:s', $data['updated_at'] );
+		$data['created_at'] = gmdate( 'Y-m-d H:i:s', $data['created_at'] );
+
+		return $data;
+	}
+
+	/**
+	 * Insert a workflow into the database.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param Automation_Workflow $workflow The workflow to persist.
+	 *
+	 * @throws Workflow_Exception Throw error if the workflow could not be inserted.
+	 */
+	protected function insert( Automation_Workflow $workflow ) {
+
+		$time = time();
+		$workflow->set_created_at( $time );
+		$workflow->set_updated_at( $time );
+
+		$data = $this->prepare_data_to_persist( $workflow );
+
+		// Technically speaking, then "id" could contain a string which
+		// would cause conflicts with the database, so we should unset
+		// it to be safe.
+		unset( $data['id'] );
+
+		$inserted = $this->wpdb->insert(
+			$this->table_name,
+			$data
+		);
+
+		if ( ! $inserted ) {
+			throw new Workflow_Exception(
+				$this->wpdb->last_error,
+				Workflow_Exception::FAILED_TO_INSERT
+			);
+		}
+
+		$workflow->set_id( $this->wpdb->insert_id );
```

Comment:
Set the ID assigned, so the workflow instance will have the changes without querying the DB.